plugins/eos/gs-plugin-eos.c

```diff
@@ -1953,6 +1990,10 @@ gs_plugin_add_distro_upgrades (GsPlugin *plugin,
 		return TRUE;
 	}
 
+	/* check if the OS upgrade has been disabled */
+	if (priv->updater_proxy == NULL)
+		return TRUE;
```

Comment:
This should be moved to before the `get_os_upgrade()` call, or the debug code path will crash as it assumes `priv->updater_proxy != NULL`

Reply:
The `get_os_upgrade()` runs after we create the proxy and only creates the GsApp for the upgrade once, so it won't crash as you say.
However, I was counting on the proxy only being `NULL` after we disable the upgrades. But if it fails to create the proxy for some reason, then yes, it would crash. I will fix that.